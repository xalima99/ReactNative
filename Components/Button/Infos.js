import React from "react";
import { View, Image, Text } from "react-native";
import styles from "./styles";
import { ClearBackGround } from "../../constants/Colors";

const Infos = ({
  children,
  icon,
  iconColor,
  textStyles,
  secondary,
  outlined,
  isPressed,
}) => {
  return (
    <View style={styles.main}>
      {icon && (
        <Image
          style={[styles.icon, { tintColor: iconColor ? iconColor : "#fff" }]}
          source={icon}
        />
      )}

      <View>
        <Text
          style={[
            textStyles,
            {
              color:
                secondary && outlined && isPressed
                  ? ClearBackGround.Fills.Default
                  : textStyles[textStyles.length - 1].color,
            },
          ]}
        >
          {children}
        </Text>
      </View>
    </View>
  );
};

export default Infos;
