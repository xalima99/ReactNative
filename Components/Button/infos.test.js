import React from "react";
import Infos from "./Infos";
import { render } from "@testing-library/react-native";

test("should render given children", () => {
  const page = render(
    <Infos
      propforward="helloWord"
      textStyles={[{ color: "red" }, { color: "red" }]}
      children="axalib"
    />
  );
  const text = page.queryByText("axalib");

  expect(text).toBeTruthy();

  console.log(page.props);
  expect(page.toJSON()).toMatchSnapshot();
});
