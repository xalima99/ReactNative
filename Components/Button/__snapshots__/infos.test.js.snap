// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render given children 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        Array [
          Array [
            Object {
              "color": "red",
            },
            Object {
              "color": "red",
            },
          ],
          Object {
            "color": "red",
          },
        ]
      }
    >
      axalib
    </Text>
  </View>
</View>
`;
